pig/src/org/partiql/pig/domain/parser/TypeDomainParser.kt

```diff
@@ -15,275 +15,393 @@
 
 package org.partiql.pig.domain.parser
 
-import com.amazon.ionelement.api.*
 import com.amazon.ion.IonReader
 import com.amazon.ion.system.IonReaderBuilder
-import org.partiql.pig.domain.model.*
-
-/** Parses a type universe contained in [universeText]. */
-fun parseTypeUniverse(universeText: String) =
-    IonReaderBuilder.standard().build(universeText).use {
-        parseTypeUniverse(it)
+import com.amazon.ionelement.api.AnyElement
+import com.amazon.ionelement.api.IonElement
+import com.amazon.ionelement.api.IonElementException
+import com.amazon.ionelement.api.IonElementLoaderOptions
+import com.amazon.ionelement.api.IonLocation
+import com.amazon.ionelement.api.MetaContainer
+import com.amazon.ionelement.api.SexpElement
+import com.amazon.ionelement.api.SymbolElement
+import com.amazon.ionelement.api.createIonElementLoader
+import com.amazon.ionelement.api.emptyMetaContainer
+import com.amazon.ionelement.api.head
+import com.amazon.ionelement.api.location
+import com.amazon.ionelement.api.metaContainerOf
+import com.amazon.ionelement.api.tag
+import com.amazon.ionelement.api.tail
+import org.partiql.pig.domain.include.IncludeCycleHandler
+import org.partiql.pig.domain.include.IncludeResolutionException
+import org.partiql.pig.domain.include.IncludeResolver
+import org.partiql.pig.domain.include.InputSource
+import org.partiql.pig.domain.model.Arity
+import org.partiql.pig.domain.model.DataType
+import org.partiql.pig.domain.model.NamedElement
+import org.partiql.pig.domain.model.PermutedDomain
+import org.partiql.pig.domain.model.PermutedSum
+import org.partiql.pig.domain.model.Statement
+import org.partiql.pig.domain.model.Transform
+import org.partiql.pig.domain.model.TupleType
+import org.partiql.pig.domain.model.TypeDomain
+import org.partiql.pig.domain.model.TypeRef
+import org.partiql.pig.domain.model.TypeUniverse
+import org.partiql.pig.errors.ErrorContext
+import org.partiql.pig.errors.PigError
+import org.partiql.pig.errors.PigException
+import java.nio.file.FileSystem
+import java.nio.file.Files
+import java.nio.file.Path
+
+/**
+ * Parses the type universe specified in [mainTypeUniversePath].
+ *
+ * @param mainTypeUnversePath Specifies the "main" type universe file where parsing will begin.  The [FileSystem]
```

Comment:
typo: mainTypeUnversePath -> mainTypeUniversePath
```suggestion
 * @param mainTypeUniversePath Specifies the "main" type universe file where parsing will begin.  The [FileSystem]
```